sdk/python/feast/feature_server.py

```diff
@@ -86,19 +88,34 @@ async def get_body(request: Request):
     def get_online_features(body=Depends(get_body)):
         try:
             body = json.loads(body)
+            full_feature_names = body.get("full_feature_names", False)
+            entity_rows = body["entities"]
             # Initialize parameters for FeatureStore.get_online_features(...) call
             if "feature_service" in body:
-                features = store.get_feature_service(
+                feature_service = store.get_feature_service(
                     body["feature_service"], allow_cache=True
                 )
+                assert_permissions(
+                    resource=feature_service, actions=[AuthzedAction.QUERY_ONLINE]
+                )
+                features = feature_service
             else:
                 features = body["features"]
-
-            full_feature_names = body.get("full_feature_names", False)
+                all_feature_views, _ = utils._get_feature_views_to_use(
+                    store.registry,
+                    store.project,
+                    features,
+                    allow_cache=True,
+                    hide_dummy_entity=False,
+                )
+                for feature_view in all_feature_views:
```

Comment:
I'm not going to do this right now, but we probably need an assert function accepting a list of resources